### PR TITLE
fix: build failure caused by luarocks connection

### PR DESCRIPTION
**Pull request by Yiyiyimu**
Signed-off-by: yiyiyimu <wosoyoung@gmail.com>

Add retry like https://github.com/apache/apisix/pull/4024

need to revert after luarocks back to normal

### CODE REVIEW

File: all-in-one/apisix-dashboard/Dockerfile [hunk 1]
```diff
@@ -22,7 +22,7 @@ RUN set -x \
     && mkdir ~/.luarocks \
     && luarocks config variables.OPENSSL_LIBDIR /usr/local/openresty/openssl/lib \
     && luarocks config variables.OPENSSL_INCDIR /usr/local/openresty/openssl/include \
-    && luarocks install https://github.com/apache/apisix/raw/master/rockspec/apisix-${APISIX_VERSION}-0.rockspec --tree=/usr/local/apisix/deps \
+    && (for (( i = 0; i < 10; i++ )); do if [[ "$i" -eq 10 ]]; then echo "failed to make deps in time"; exit 1; fi; luarocks install https://github.com/apache/apisix/raw/master/rockspec/apisix-${APISIX_VERSION}-0.rockspec --tree=/usr/local/apisix/deps && break; done) \
```
Comment (spacewander) [inline all-in-one/apisix-dashboard/Dockerfile:25]: "failed to make deps in time" should be "failed to fetch Lua rocks in time"?

Comment (Yiyiyimu) [reply all-in-one/apisix-dashboard/Dockerfile]: Thanks for remind! Fixed.

File: all-in-one/apisix-dashboard/Dockerfile [hunk 2]
```diff
@@ -22,7 +22,7 @@ RUN set -x \
     && mkdir ~/.luarocks \
     && luarocks config variables.OPENSSL_LIBDIR /usr/local/openresty/openssl/lib \
     && luarocks config variables.OPENSSL_INCDIR /usr/local/openresty/openssl/include \
-    && (for (( i = 0; i < 10; i++ )); do if [[ "$i" -eq 10 ]]; then echo "failed to make deps in time"; exit 1; fi; luarocks install https://github.com/apache/apisix/raw/master/rockspec/apisix-${APISIX_VERSION}-0.rockspec --tree=/usr/local/apisix/deps && break; done) \
+    && (for (( i = 0; i < 10; i++ )); do if [[ "$i" -eq 10 ]]; then echo "failed to fetch luarocks in time"; exit 1; fi; luarocks install https://github.com/apache/apisix/raw/master/rockspec/apisix-${APISIX_VERSION}-0.rockspec --tree=/usr/local/apisix/deps && break; done;) \
```
Comment (spacewander) [inline all-in-one/apisix-dashboard/Dockerfile]: Emm... We failed to fetch Lua rocks but not the luarocks (the tool)

Comment (Yiyiyimu) [reply all-in-one/apisix-dashboard/Dockerfile]: Ah I get it! Thanks for pointing it out!!

**Review by tokers — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**